src/OrchardCore.Modules/OrchardCore.Roles/Migrations/SystemRolesMigrations.cs

```diff
@@ -0,0 +1,40 @@
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Logging;
+using OrchardCore.Data.Migration;
+using OrchardCore.Security;
+
+namespace OrchardCore.Roles.Migrations;
+
+public sealed class SystemRolesMigrations : DataMigration
+{
+    private readonly ISystemRoleNameProvider _systemRoleNameProvider;
+    private readonly RoleManager<IRole> _roleManager;
+    private readonly ILogger _logger;
+
+    public SystemRolesMigrations(
+        ISystemRoleNameProvider systemRoleNameProvider,
+        RoleManager<IRole> roleManager,
+        ILogger<RolesMigrations> logger)
+    {
+        _systemRoleNameProvider = systemRoleNameProvider;
+        _roleManager = roleManager;
+        _logger = logger;
+    }
+
+    public async Task<int> CreateAsync()
+    {
+        var systemRoles = await _systemRoleNameProvider.GetSystemRolesAsync();
+
+        foreach (var role in systemRoles)
+        {
+            if (await _roleManager.FindByNameAsync(role) is null)
+            {
+                await _roleManager.CreateAsync(new Role { RoleName = role });
```

Comment:
The descriptions of the roles need to be retained too.

Reply:
I already pointed to that in https://github.com/OrchardCMS/OrchardCore/pull/17341#issuecomment-2585976800

Reply:
Well, think about this, please.

Reply:
> @MikeAlhayek I didn't see a description there, or shall we make ISystemRoleProvider instead of ISystemRoleNameProvider?

@hishamco yes you can do that. But, Don't remove `ISystemRoleNameProvider`, you can obsolete it. 

I suggest the new interface to look like this

```
public interface ISystemRoleProvider
{
    ValueTask<IRole> GetAdminRoleAsync();

    ValueTask<bool> IsSystemRoleAsync(string name);

    ValueTask<IEnumerable<IRole>> GetSystemRolesAsync();
}
```

Then you can add this extension
https://github.com/OrchardCMS/OrchardCore/blob/c42af0f5609a903e93fb2ac1b813454b8508827c/src/OrchardCore/OrchardCore.Roles.Core/SystemRoleNameProviderExtensions.cs#L5 

for `ISystemRoleProvider`

Reply:
I thought about adding such thing, but it's great to hear your feedback